src/pages/inquiry/index.tsx

```diff
@@ -0,0 +1,112 @@
+import { InquiryApi, InquiryData } from '@/api'
+import { ArrowBlock } from '@/components/ui'
+import BaseLayout from '@/layouts/BaseLayout'
+import Grid from '@/layouts/Grid'
+import { useStateRef } from '@/modules/use-state-ref'
+import { NextPage } from 'next'
+import Head from 'next/head'
+import Router from 'next/router'
+import React, { useEffect } from 'react'
+import './InquiryPage.scss'
+type InquiryProps = {
+  inquiries: InquiryData[]
+  isAdmin: boolean
+}
+
+function formatDate(date): string {
+  return (
+    date.getFullYear() +
+    '-' +
+    ('0' + (date.getMonth() + 1)).slice(-2) +
+    '-' +
+    ('0' + date.getDate()).slice(-2) +
+    ' ' +
+    ('0' + date.getHours()).slice(-2) +
+    ':' +
+    ('0' + date.getMinutes()).slice(-2) +
+    ':' +
+    ('0' + date.getSeconds()).slice(-2)
+  )
+}
+const InquiryPage: NextPage<InquiryProps> = ({
+  inquiries: inquiries,
+  isAdmin,
+}) => {
+  const [items, setItems, itemsRef] = useStateRef(inquiries)
+  const [isFetching, setIsFetching, isFetchRef] = useStateRef(false)
+
+  function isReachBottom(): boolean {
+    return window.innerHeight < document.body.scrollHeight
+  }
+
+  async function loadMore(): Promise<void> {
+    if (isFetchRef.current) {
+      return
+    }
+    setIsFetching(true)
+    const moreItems = await InquiryApi.inquirys(itemsRef.current.length)
+    if (moreItems.inquiries && moreItems.inquiries.length > 0) {
+      setItems([...itemsRef.current, ...moreItems.inquiries])
+    }
+    setIsFetching(false)
+  }
+
+  useEffect(() => {
+    if (inquiries === undefined) {
+      Router.push('/inquiry/request')
```

Comment:
Redirections should be processed on the server side for better SEO.
Use `getInitialProps`.